pkg/handlers/nutanix/mutation/machinedetails/inject.go

```diff
@@ -94,7 +94,11 @@ func (h *nutanixMachineDetailsPatchHandler) Mutate(
 
 			spec.BootType = nutanixMachineDetailsVar.BootType
 			spec.Cluster = nutanixMachineDetailsVar.Cluster
-			spec.Image = nutanixMachineDetailsVar.Image
+			if nutanixMachineDetailsVar.Image != nil {
+				spec.Image = nutanixMachineDetailsVar.Image.DeepCopy()
+			} else if nutanixMachineDetailsVar.ImageLookup != nil {
+				spec.ImageLookup = nutanixMachineDetailsVar.ImageLookup.DeepCopy()
+			}
```

Comment:
I know this is protected by the CEL validation in API definition but let's be defensive here, using a `switch` and return error if not set or both set.